platform.js

```diff
@@ -423,5 +423,16 @@ module.exports = {
         });
       }
     });
+  },
+  runPromise(promise, retries) {
+    return promise()
+      .catch((err)=>{
+        if(retries > 0) {
+          return module.exports.runPromise(promise, retries - 1);
```

Comment:
@ahmedalsudani makes sense. I'll add it as an optional parameter. Should we retry on timeouts or just give up?